icu4c/source/common/dictbe.cpp

```diff
@@ -1385,6 +1405,26 @@ CjkBreakEngine::divideUpDictionaryRange( UText *inText,
     // inputMap goes out of scope
     return numBreaks;
 }
+
+void CjkBreakEngine::loadJpParticleAndAuxVerbs() {
+    UErrorCode error = U_ZERO_ERROR;
+    const char* tags[2] = {"particles", "auxVerbs"};
+
+    ResourceBundle ja(U_ICUDATA_BRKITR, "ja", error);
```

Comment:
so, in the way you read the resource, you get the UnicodeString from the resources, and then put into fParticleSet, which is a Hashtable and you always put in true for that. I think you should call fParticleSet.puti(str, 1, error);  instead, which will save some memory since the value is always number.

Reply:
Resolved